rizinorg/rizin
# Fix rz_path_prefix not working when BINDIR is at the root directory or several folders deep

**Your checklist for this pull request**

**Detailed description**

Fix `rz_path_prefix` not working when BINDIR is at the root directory or several folders deep.

This change aims to solve [#2897](https://github.com/rizinorg/cutter/issues/2897) - the Cutter build artifact fails to find the themes directory due to changes in [rz_path_prefix](https://github.com/rizinorg/rizin/blob/dev/librz/util/path.c#L25), which only returns the relative prefix for portable binaries that return true for `rz_str_endswith(<pid path>, RZ_SYS_DIR RZ_BINDIR)`. Since Windows Cutter sets `RZ_BINDIR=.` and Linux Cutter runs from the root directory while `RZ_BINDIR=bin`, the original build directory prefix is returned in all cases.

**Test plan**

- Check that Rizin is able to load themes with `-Dbindir=multiple/prefixes`
- Check that Rizin is able to load themes with `-Dbindir=.`
- Verify that [#2897](https://github.com/rizinorg/cutter/issues/2897) is solved by these changes.

**Closing issues**



...
